pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -294,29 +294,10 @@ func (a Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSpe
 		return err
 	}
 
-	objectMeta := kclient.CreateObjectMeta(componentName, a.Client.Namespace, labels, nil)
-	podTemplateSpec := kclient.GeneratePodTemplateSpec(objectMeta, containers)
-
-	kclient.AddBootstrapSupervisordInitContainer(podTemplateSpec)
-
-	// if there are preStart events, add them as init containers to the podTemplateSpec
-	preStartEvents := a.Devfile.Data.GetEvents().PreStart
-	if len(preStartEvents) > 0 {
-		var eventCommands []string
-		commandsMap := a.Devfile.Data.GetCommands()
-		containersMap := utils.GetContainersMap(containers)
-
-		for _, event := range preStartEvents {
-			eventSubCommands := common.GetCommandsFromEvent(commandsMap, strings.ToLower(event))
-			eventCommands = append(eventCommands, eventSubCommands...)
-		}
-
-		klog.V(4).Infof("PreStart event commands are: %v", strings.Join(eventCommands, ","))
-		utils.AddPreStartEventInitContainer(podTemplateSpec, commandsMap, eventCommands, containersMap)
-		if len(eventCommands) > 0 {
-			log.Successf("PreStart commands have been added to the component: %s", strings.Join(eventCommands, ","))
-		}
-	}
+	objectMeta := generator.CreateObjectMeta(componentName, a.Client.Namespace, labels, nil)
+	supervisordInitContainer := kclient.AddBootstrapSupervisordInitContainer()
+	initContainers := utils.GetPreStartInitContainers(a.Devfile, containers)
```

Comment:
why you need to pass containers to get init containers?

Reply:
`containers` is all the component containers returned from the devfile. PreStart events reference a command(which in turn reference a container component). So we scan thru all the containers that we have and append to `initContainers`